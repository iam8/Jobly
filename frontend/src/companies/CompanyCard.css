.CompanyCard .card {
    min-height: 10vh;
}

.CompanyCard a {
    color: black;
    text-decoration: none;
}
