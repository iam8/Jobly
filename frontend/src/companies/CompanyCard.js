import React from "react";
import { Link } from "react-router-dom";
import {Card, CardTitle, CardBody, CardText, CardImg, Row, Col} from "reactstrap";

import "./CompanyCard.css";


/** Display basic information about a company. */
function CompanyCard({name, desc, logoUrl, handle}) {

    return (
        <div className="CompanyCard">
            <Card>
                <CardBody>
                    <Link to={`/companies/${handle}`}>
                        <Row>
                            <Col>
                                <CardTitle tag="h5">{name}</CardTitle>
                                <CardText>{desc}</CardText>
                            </Col>

                            <Col id="CompanyCard-img" className="col-2">
                                {logoUrl &&
                                    <CardImg
                                        tag="img"
                                        src={logoUrl}
                                        alt={name}
                                    />
                                }
                            </Col>
                        </Row>
                    </Link>
                </CardBody>
            </Card>
        </div>
    )
}


export default CompanyCard;
