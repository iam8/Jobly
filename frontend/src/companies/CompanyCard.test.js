import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";

import CompanyCard from "./CompanyCard";


test("Renders without crashing", () => {
    render(
        <MemoryRouter>
            <CompanyCard
                handle="test-0"
                name="Test Company"
                description="Desc 0"
                logo_url="dummyUrl.jpg"
            />
        </MemoryRouter>
    )
})


test("Matches snapshot with image", () => {
    const { asFragment } = render(
        <MemoryRouter>
            <CompanyCard
                handle="test-0"
                name="Test Company"
                description="Desc 0"
                logoUrl="dummyUrl.jpg"
            />
        </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
});


test("matches snapshot without image", () => {
    const { asFragment } = render(
        <MemoryRouter>
            <CompanyCard
                handle="test-0"
                name="Test Company"
                description="Desc 0"
            />
        </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
});
