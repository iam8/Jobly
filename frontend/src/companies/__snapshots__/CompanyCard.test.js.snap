// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot with image 1`] = `
<DocumentFragment>
  <div
    class="CompanyCard"
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <a
          href="/companies/test-0"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <h5
                class="card-title"
              >
                Test Company
              </h5>
              <p
                class="card-text"
              />
            </div>
            <div
              class="col-2 col"
              id="CompanyCard-img"
            >
              <img
                alt="Test Company"
                class="card-img"
                src="dummyUrl.jpg"
              />
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot without image 1`] = `
<DocumentFragment>
  <div
    class="CompanyCard"
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <a
          href="/companies/test-0"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <h5
                class="card-title"
              >
                Test Company
              </h5>
              <p
                class="card-text"
              />
            </div>
            <div
              class="col-2 col"
              id="CompanyCard-img"
            />
          </div>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
